client/app/components/dashboards/AddTextboxDialog.jsx

```diff
@@ -0,0 +1,152 @@
+import { markdown } from 'markdown';
+import { debounce } from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { react2angular } from 'react2angular';
+
+class AddTextboxDialog extends React.Component {
+  static propTypes = {
+    dashboard: PropTypes.object, // eslint-disable-line react/forbid-prop-types
+    close: PropTypes.func,
+    dismiss: PropTypes.func,
+  };
+
+  static defaultProps = {
+    dashboard: null,
+    close: () => {},
+    dismiss: () => {},
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      saveInProgress: false,
+      text: '',
+      preview: '',
+    };
+
+    const updatePreview = debounce(() => {
+      this.setState({
+        preview: markdown.toHTML('' + this.state.text),
```

Comment:
Why do we need `'' + ..` here?

Reply:
Also why preview is a state property and not computed when rendering? To have `debounce`?

Reply:
> Why do we need '' + .. here?

Not sure. Will remove.

> Also why preview is a state property and not computed when rendering? To have debounce?

Yes. Otherwise it may lag a bit when typing in textarea